bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -730,15 +724,37 @@ class Plot extends Component.Model
       attrs['renderers'] = _.filter(attrs['renderers'], (r) -> r.serializable_in_document())
     attrs
 
-  mixins: [
-    'text:title_',
-    'line:outline_',
-    'fill:border_',
-    'fill:background_'
-  ]
+  add_renderers: (new_renderers...) ->
+    renderers = @get('renderers')
+    renderers = renderers.concat(new_renderers)
+    @set('renderers', renderers)
 
-  props: ->
-    return _.extend {}, super(), {
+  add_layout: (renderer, place="center") ->
+    if renderer.props.plot?
+      renderer.plot = this
+
+    @add_renderers(renderer)
+
+    if place != 'center'
+      @set(place, @get(place).concat([renderer]))
+
+  add_glyph: (glyph, source) ->
+    if not source?
+      source = new ColumnDataSource.Model()
+
+    renderer = new GlyphRenderer.Model({data_source: source, glyph: glyph})
+    @add_renderers(renderer)
+
+    return renderer
+
+  add_tools: (tools...) ->
+    for tool in tools
+      tool.plot = this
+      @set("tools", @get("tools").concat([tool]))
```

Comment:
mmm, also, tools seem to need to have the plot object at instantiation (at least with current master).